src/core/src/common/errors.rs

```diff
@@ -1,7 +1,22 @@
-#[derive(Debug, PartialEq)]
-pub enum Error {
-    ParseError(std::num::ParseIntError),
+use anyhow::Error;
+
+#[derive(thiserror::Error, Debug)]
+pub enum CoreError {
+    #[error("parse error {0}")]
+    ParseError(#[from] std::num::ParseIntError),
+
+    #[error("io error {0}")]
+    IOError(#[from] std::io::Error),
+
+    #[error("missing parameters")]
     MissingParameters,
+
+    #[error("not found")]
     NotFound,
-    InternalError,
+
+    #[error("internal error {0}")]
+    InternalError(#[from] Error),
+
+    #[error("unknown data store error")]
```

Comment:
unknown data store error ?

Reply:
I'll remove this `Unknown` error because it is the same thing with `Internal Error`